watertap/examples/flowsheets/RO_with_energy_recovery/parameter_sweep_input_parser.py

```diff
@@ -0,0 +1,310 @@
+                            try:
+                                key.append(float(idx))
+                            except:
+                                key.append(idx)
+
+                        key = tuple(key)
+
+                    else:
+                        print('Child indexing type %s not recognized.' % (index_type))
+
+                else:
+                    # We have an indexed child, but no bracketed index was found in the string specifier.
+                    # Iterate through the child's keys and raise an error if >1 options exist.
+                    # If only one option exists, use it as the key to descend further along the path. 
+                    ct = 0
+                    for key in child.keys():
+                        ct += 1
+
+                    if ct > 1:
+                        raise ValueError('No bracketed index found for %s path specifier but multiple options exist.')
+
+                child = child[key]
+
+            parent = child
+
+    return child
+
+
+def get_sweep_params_from_yaml(m, yaml_filename):
+    """ Creates a dictionary of swept model parameters specified via yaml file
+
+    This function creates a dictionary of the items to vary during a parameter
+    sweep where the variable name, model attribute, and sweeping domain are
+    specified in a YAML file.  The YAML file should have the following format::
+
+        A_comp:
+            type: NormalSample
+            model_value: m.fs.RO.A_comp
+            mean: 4.0e-12
+            std: 0.5e-12
+
+    where the top-level keyword can be any short, easily understood identifier
+    for the parameter.  ``type`` must be one of ``LinearSample``, ``UniformSample``,
+    ``NormalSample``, or ``LatinHypercubeSample``.  ``model_value`` must be a valid
+    dot-sperated string path to the object attribute (in this case, an RO attribute
+    on the flowsheet ``m``) that you wish to vary.  The remaining arguments are
+    dependent on the sample type selected.  For ``NormalSample`` information about
+    the mean and standard deviation is required.  Consult the ``parameter_sweep``
+    help for more information on the different sample classes.
+
+    Args:
+        m (pyomo model):
+            The flowsheet containing the model to deploy with the parameter sweep
+            tool.
+        yaml_filename (str):
+            The path to the yaml file.
+
+    Returns:
+        sweep_params (dict):
+            A dictionary containing different instances of parameter sweep samples
+
+    """
+
+    input_dict = _yaml_to_dict(yaml_filename)
+
+    sweep_params = {}
+
+    for param, values in input_dict.items():
+
+
+        # Split the dot-separated object path into a list of
+        # separate objects and attributes
+        path = _split_pyomo_path(values['model_value'])
+
+        # Follow the attribute path to the correct nested
+        # property, should work with indexing!
+        child = _return_child_property(m, path)
```

Comment:
If you drop the `m.` in the name, you can replace these lines (and both functions) with
```
child = m.find_component(values['model_value'])
if child is None:
    raise ValueError(f"Could not acccess attribute {values['model_value']}")

Reply:
Didn't know about this, definitely much easier--thanks Ben.